docs/docs/concepts/memory.md

```diff
@@ -190,94 +194,73 @@ item = store.get(namespace, "a-memory")
 
-**Do you want to manage memories as a single profile or as a collection of documents?**
+**What is the type of memory?**
 
-We provide two main approaches to managing long-term memory: a single, continuously updated document (referred to as a "profile" or "schema") or a collection of documents. Each method offers its own benefits, depending on the type of information you need to store and how you intend to access it.
+Humans use memories to remember [facts](https://en.wikipedia.org/wiki/Semantic_memory), [experiences](https://en.wikipedia.org/wiki/Episodic_memory), and [rules](https://en.wikipedia.org/wiki/Procedural_memory). AI agents can use memory in the same ways. For example, AI agents can use memory to remember specific facts about a user to accomplish a task. We expand on several types of memories in the [section below](#memory-types).
 
-Managing memories as a single, continuously updated "profile" or "schema" is useful when there is well-scoped, specific information you want to remember about a user, organization, or other entity (including the agent itself). You can define the schema of the profile ahead of time, and then use an LLM to update this based on interactions. Querying the "memory" is easy since it's a simple GET operation on a JSON document. We explain this in more detail in [remember a profile](#manage-individual-profiles). This technique can provide higher precision (on known information use cases) at the expense of lower recall (since you have to anticipate and model your domain, and updates to the doc tend to delete or rewrite away old information at a greater frequency).
+![](img/memory/memory_types.png)
 
-Managing long-term memory as a collection of documents, on the other hand, lets you store an unbounded amount of information. This technique is useful when you want to repeatedly extract & remember items over a long time horizon but can be more complicated to query and manage over time.
-Similar to the "profile" memory, you still define schema(s) for each memory. Rather than overwriting a single document, you instead will insert new ones (and potentially update or re-contextualize existing ones in the process). We explain this approach in more detail in ["managing a collection of memories"](#manage-a-collection-of-memories).
+**When do you want to update memories?**
 
-**Do you want to present memories to your agent as updated instructions or as few-shot examples?**
+Memory can be updated as part of an agent's application logic (e.g. "on the hot path"). In this case, the agent typically decides to remember facts before responding to a user. Alternatively, memory can be updated as a background task (as a separate function that generates memories based on the primary application's state). We explain the tradeoffs between these approaches in the [section below](#writing-memories).
```

Comment:
nit: "a separate function" a bit strange in terms of terminoloy? what doe it mean that it's a "function"? does it have to be a function or can i provide it as a method or a class? what does it mean that it's "separate"?

Maybe could paraphrase a bit to be more general "some logic that runs in the background / asynchronously and generates memories" ?

Reply:
> Maybe could paraphrase a bit to be more general "some logic that runs in the background / asynchronously and generates memories" ?

yes, function is too specific. will update.